Nos/Service/NostrIdentifier/TLVElement.swift

```diff
@@ -3,17 +3,17 @@ import Foundation
 /// A TLV (type-length-value) element, which represents a single type, length, and value.
 /// - Note: See [NIP-19](https://github.com/nostr-protocol/nips/blob/master/19.md) for more information.
 enum TLVElement {
-    /// The special type with its associated value as a String.
-    case special(value: String)
-    
-    /// The relay type with its associated value as a String.
-    case relay(value: String)
+    /// The special type with its associated value.
+    case special(value: Data)
 
-    /// The author type with its associated value as a String.
-    case author(value: String)
-    
-    /// The kind type with its associated value as a UInt32.
-    case kind(value: UInt32)
+    /// The relay type with its associated value.
+    case relay(value: Data)
+
+    /// The author type with its associated value.
+    case author(value: Data)
+
+    /// The kind type with its associated value.
+    case kind(value: Data)
```

Comment:
Since the `value` for every case is `Data` now, I did consider switching to a `struct` from the `enum`. If we want to go that route, I'm happy to do it but I wasn't sure how important that was. And I haven't completely thought through how it would change the way we use these; maybe it would be worse at the call site.

Reply:
I'll do this. I also hope to eliminate TLVType when I finally get to #1294.

Reply:
Oh, whoops, I mean I hope to eliminate TLV. I'll be keeping TLVType. But either way, changing TLVElement to a struct should be nicer.

Reply:
Done!